### PR TITLE
Coding style: try to make a function to only one exit point and don't use goto

### PR DESCRIPTION
**Pull request by lifeix**
This is a simple demo. If this could work for almost the case, will apply it.
refine strnlen_s/strstr_s to only one exit point
timer: fix procedure has more than one exit point
remove goto in ept_violation_vmexit_handler

Tracked-On: #861
Signed-off-by: Li, Fei1 fei1.li@intel.com

### CODE REVIEW

File: hypervisor/arch/x86/timer.c [hunk 1]
```diff
@@ -64,40 +66,39 @@ static void local_add_timer(struct per_cpu_timers *cpu_timer,
...
-	bool need_update;
+	int32_t ret = 0;
 
 	if ((timer == NULL) || (timer->func == NULL) || (timer->fire_tsc == 0UL)) {
-		return -EINVAL;
+		ret = -EINVAL;
 	}
 
-	ASSERT(list_empty(&timer->node), "add timer again!\n");
+	if (ret == 0) {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/timer.c]: here use "else" is more straight forward and fast.

File: hypervisor/arch/x86/ept.c [hunk 2]
```diff
@@ -115,46 +115,44 @@ int32_t ept_violation_vmexit_handler(struct acrn_vcpu *vcpu)
...
 	 * emulation at first.
 	 */
 
 	/* Determine value being written. */
-	if (mmio_req->direction == REQUEST_WRITE) {
-		status = emulate_instruction(vcpu);
-		if (status != 0) {
-			goto out;
+	if (ret > 0) {
+		if (mmio_req->direction == REQUEST_WRITE) {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/ept.c]: move the following code right after Line 117?

Comment (lifeix) [reply hypervisor/arch/x86/ept.c]: Done

File: hypervisor/arch/x86/timer.c (hunk 1 above)
Comment (lifeix) [reply hypervisor/arch/x86/timer.c]: Done

**Comment by HuiHuangShi** [comment]
2057<br>1 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/lib/string.c&nbsp;&nbsp;function:strstr_s&nbsp;&nbsp;offset:393:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Cast on a constant value.                    : pret<br><br>

**Pull request merged**